lib/create-testing-library-rule/detect-testing-library-utils.ts

```diff
@@ -0,0 +1,917 @@
+   * related module.
+   */
+  const findImportedUtilSpecifier: FindImportedUtilSpecifierFn = (
+    specifierName
+  ) => {
+    const node = getCustomModuleImportNode() ?? getTestingLibraryImportNode();
+
+    if (!node) {
+      return undefined;
+    }
+
+    if (isImportDeclaration(node)) {
+      const namedExport = node.specifiers.find((n) => {
+        return (
+          isImportSpecifier(n) &&
+          [n.imported.name, n.local.name].includes(specifierName)
+        );
+      });
+
+      // it is "import { foo [as alias] } from 'baz'""
+      if (namedExport) {
+        return namedExport;
+      }
+
+      // it could be "import * as rtl from 'baz'"
+      return node.specifiers.find((n) => isImportNamespaceSpecifier(n));
+    } else {
+      if (!ASTUtils.isVariableDeclarator(node.parent)) {
+        return undefined;
+      }
+      const requireNode = node.parent;
+
+      if (ASTUtils.isIdentifier(requireNode.id)) {
+        // this is const rtl = require('foo')
+        return requireNode.id;
+      }
+
+      // this should be const { something } = require('foo')
+      if (!isObjectPattern(requireNode.id)) {
+        return undefined;
+      }
+
+      const property = requireNode.id.properties.find(
+        (n) =>
+          isProperty(n) &&
+          ASTUtils.isIdentifier(n.key) &&
+          n.key.name === specifierName
+      );
+      if (!property) {
+        return undefined;
+      }
+      return (property as TSESTree.Property).key as TSESTree.Identifier;
+    }
+  };
+
+  const findImportedUserEventSpecifier: () => TSESTree.Identifier | null = () => {
+    if (!importedUserEventLibraryNode) {
+      return null;
+    }
+
+    if (isImportDeclaration(importedUserEventLibraryNode)) {
+      const userEventIdentifier = importedUserEventLibraryNode.specifiers.find(
+        (specifier) => isImportDefaultSpecifier(specifier)
+      );
+
+      if (userEventIdentifier) {
+        return userEventIdentifier.local;
+      }
+    } else {
+      if (!ASTUtils.isVariableDeclarator(importedUserEventLibraryNode.parent)) {
+        return null;
+      }
+
+      const requireNode = importedUserEventLibraryNode.parent;
+      if (!ASTUtils.isIdentifier(requireNode.id)) {
+        return null;
+      }
+
+      return requireNode.id;
+    }
+
+    return null;
+  };
+
+  const getImportedUtilSpecifier = (
+    node: TSESTree.MemberExpression | TSESTree.Identifier
+  ): TSESTree.ImportClause | TSESTree.Identifier | undefined => {
+    const identifierName: string | undefined = getPropertyIdentifierNode(node)
+      ?.name;
+
+    if (!identifierName) {
+      return undefined;
+    }
+
+    return findImportedUtilSpecifier(identifierName);
+  };
+
+  /**
+   * Determines if file inspected meets all conditions to be reported by rules or not.
+   */
+  const canReportErrors: CanReportErrorsFn = () => {
+    return skipRuleReportingCheck || isTestingLibraryImported();
```

Comment:
Part 3

Reply:
Oh, I see you already did the change. This is even better than where I suggested putting the check! Moving the file added a lot of noise tho, please let's revert that.